src/lib/comms/sol-network-impl-linux.c

```diff
@@ -75,6 +75,68 @@ struct sol_network {
 
 static struct sol_network *network = NULL;
 
+static struct sol_network_link_addr *
+bluetooth_addr_from_str(struct sol_network_link_addr *addr, const char *str)
+{
+    int i;
+    uint8_t *ba = addr->addr.bt_addr;
+
+    /* FIXME */
+    addr->family = SOL_NETWORK_FAMILY_BLUETOOTH;
+    addr->addr.bt_type = SOL_NETWORK_BT_ADDR_BASIC_RATE;
+
+    for (i = 5; i >= 0; i--, str += 3)
```

Comment:
you should iterate str based on endptr from strtol... and it should be the strtoul (unsigned)